### PR TITLE
Saithriftv2 fixes for extensions

### PR DESCRIPTION
**Pull request by chrispsommers**
Fixes: https://github.com/opencomputeproject/SAI/issues/1529, https://github.com/opencomputeproject/SAI/issues/1532
I will follow up with edits to usage guide (pending: https://github.com/opencomputeproject/SAI/pull/1512) once it's merged.

### CODE REVIEW

File: test/saithriftv2/Makefile [hunk 1]
```diff
@@ -88,7 +92,7 @@ $(ODIR)/sai_rpc_server.o: ../../meta/sai_rpc_frontend.cpp
 	$(CXX) $(CPPFLAGS) -c $^ -o $@ $(CPPFLAGS) -I./gen-cpp -I../../meta -I../../inc -I../../experimental
 
 $(ODIR)/saiserver.o: src/saiserver.cpp $(CPP_SOURCES) directories
-	$(CXX) $(CPPFLAGS) -c src/saiserver.cpp  -o $@ $(CPPFLAGS) $(CDEFS) -I./gen-cpp -I../../inc
+	$(CXX) $(CPPFLAGS) -c src/saiserver.cpp  -o $@ $(CPPFLAGS) $(CDEFS) -I./gen-cpp -I../../inc -I../../experimental
```
Comment (vmytnykx) [inline test/saithriftv2/Makefile:95]: btw, this looks like an issue, as some part of the code is generated based on in tree headers and some are taken from /usr/include/sai. I think this needs to be fixed some time, not part of this change.

**Review by vmytnykx — COMMENTED** [review]
This needs to be ported to dash branch once it is merged into master.

File: test/saithriftv2/Makefile [hunk 2]
```diff
@@ -8,16 +8,20 @@ METADIR := $(PREDIR)$(PREDIR)meta/
 
 SAI_PREFIX = /usr
 SAI_HEADER_DIR ?= $(SAI_PREFIX)/include/sai
+SAI_EXT_HEADER_DIR ?= $(SAI_PREFIX)/include/experimental
```
[the remark this answers is not in the document]
Comment (chrispsommers) [reply test/saithriftv2/Makefile:11]: I'd agree, but the structure of all the headers was already established and I just found paths to them. My intuition would have made it something like /user/include/sai/[inc,experimental] but that's not how it is. Instead the headers in SAI/inc are copied to /usr/include/sai.

File: test/saithriftv2/Makefile (hunk 1 above)
Comment (chrispsommers) [reply test/saithriftv2/Makefile:95]: If you think so, it might be good to file an issue. However, a later change could break automated workflows which use the current technique.

File: test/saithriftv2/Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (vmytnykx) [reply test/saithriftv2/Makefile:11]: What I mean here is that we need to be able to generate saiserver based on dash SAI branch that includes DASH experimental API (overlay). For this to work, we need to backport this PR to dash branch and generate the saiserver from there.

**Review by kcudnik — APPROVED** [review]

File: test/saithriftv2/Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (richardyu-ms) [reply test/saithriftv2/Makefile:11]: > I'd agree, but the structure of all the headers was already established and I just found paths to them. My intuition would have made it something like /user/include/sai/[inc,experimental] but that's not how it is. Instead the headers in SAI/inc are copied to /usr/include/sai.

looks like place the SAI_EXT_HEADER_DIR  as $(SAI_PREFIX)/include/**sai**/experimental is better. 
cause there are already some code reference sai header from $(SAI_PREFIX)/include/sai, how about just keep the existing header file as it.

**Comment by richardyu-ms** [comment]
@chrispsommers please resolve the merge conflicts

**Comment by chrispsommers** [comment]
> @chrispsommers please resolve the merge conflicts

Done, thanks for the review.
